src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -254,3 +253,45 @@ class JaxArrayField(_BaseNdArrayField):
         array_ns: ClassVar[ModuleType] = jnp
 
     common.field.register(jnp.ndarray, JaxArrayField.from_array)
+
+
+def _slice_with_domain(field: common.Field, new_domain: common.Domain) -> np.ndarray:
```

Comment:
```suggestion
def _slice_with_domain(field: common.Field, new_domain: common.Domain) -> tuple[slice| new_axis, ...]:
```
maybe like this to simplify testing